daemon/config/config.go

```diff
@@ -191,9 +196,7 @@ func (cfg *Config) MergeConfigurations(flagSet *pflag.FlagSet) error {
 	}
 
 	// merge configurations from command line flags and config file
-	err = mergeConfigurations(fileConfig, cfg.delValue(flagSet, fileFlags))
```

Comment:
Why split `return mergeConfigurations` into `err = ... ; return err` before?

Reply:
because the caller will check the error, no need to check before return